src/hash.c

```diff
@@ -84,6 +86,68 @@ FILE *hash_log;
 
 /* Hash algorithms */
 
+void MD5InitGeneric(void * mdContext) {
+    MD5Init((MD5_CTX *)mdContext);
+}
+
+void MD5UpdateGeneric(void * mdContext, const void * inBuf, size_t inLen) {
+    assert(inLen <= (size_t)UINT_MAX);
+    MD5Update((MD5_CTX *)mdContext, (const unsigned char *)inBuf, (unsigned int)inLen);
+}
+
+void MD5FinalGeneric(void * mdContext, void * buf) {
+    MD5Final((MD5_CTX *)mdContext, (char *)buf);
+}
+
+void SHA1InitGeneric(void * sc) {
+    SHA1Init((SHA1Context *)sc);
+}
+
+void SHA1UpdateGeneric(void * sc, const void * vdata, size_t len) {
+    assert(len <= (size_t)0xFFFFFFFFu);
```

Comment:
I know it is a bit pedantic, and it is going to be the same, but could we use UINT32_MAX (from `stdint.h`) instead of the literal? 😨

Reply:
I will merge it as it is, and we can change it in a later commit.

Reply:
@davidpolverari I only avoided `UINT32_MAX` because it seems non-standard, e.g. I do not see it at https://pubs.opengroup.org/onlinepubs/009695399/basedefs/stdint.h.html but would expect mention there.